jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -503,6 +503,51 @@ static int fastIdentityHashCode(Object anObject) {
 	 */
 	static native int identityHashCode(Object anObject);
 
+	/*[IF INLINE-TYPES] */
+	/**
+	 * Answers an integer hash code for the parameter.
+	 * The caller must ensure that the parameter is a value type.
+	 * The hash code returned is the same one that would
+	 * be returned by java.lang.Object.hashCode(), assuming
+	 * the object's class has not overridden hashCode() and
+	 * that the parameter is a value type.
+	 *
+	 * @param		anObject	the object
+	 * @return		the hash code for the object
+	 * @throws		InternalError if the object is not a value type
+	 *
+	 * @see			java.lang.Object#hashCode
+	 */
+	static int valueHashCode(Object anObject) {
+		int hashcode = 0;
+
+		try {
+			Class ValueObjectMethods = Class.forName("java.lang.runtime.ValueObjectMethods");
+			Method valueObjectHashCode = ValueObjectMethods.getDeclaredMethod("valueObjectHashCode", new Class[]{ Object.class });
+			valueObjectHashCode.setAccessible(true);
+
+			Object[] args = new Object[]{ anObject };
+			hashcode = (Integer)valueObjectHashCode.invoke(null, args);
+		} catch (ClassNotFoundException|NoSuchMethodException|IllegalAccessException|InvocationTargetException e) {
+			throw new InternalError(e);
+		}
+
+		if (positiveOnlyHashcodes()) {
+			hashcode = Math.abs(hashcode);
+		}
+
+		return hashcode;
+	}
+
+	/**
+	 * Returns true if hash codes are positive only,
+	 * false otherwise
```

Comment:
Please finish this sentence with a period.

Reply:
updated in 0ed4550